python/pyarrow/tests/test_parquet.py

```diff
@@ -2028,7 +2028,7 @@ def test_filters_invalid_pred_op(tempdir, use_legacy_dataset):
                                     use_legacy_dataset=use_legacy_dataset)
         assert dataset.read().num_rows == 0
 
-    with pytest.raises(ValueError if use_legacy_dataset else TypeError):
+    with pytest.raises(ValueError if use_legacy_dataset else pa.ArrowInvalid):
         # dataset API returns TypeError when trying create invalid comparison
```

Comment:
```suggestion
    with pytest.raises(ValueError):
```

ArrowInvalid is a ValueError, I think, so if this changed, the above should be sufficient

Reply:
Nice, updated.